pkg/triggerconfig/inrepo/params.go

```diff
@@ -22,18 +22,30 @@ type UseLocation struct {
 
 // UseParametersAndResults adds the parameters from the used Task to the PipelineSpec if specified and the PipelineTask
 func UseParametersAndResults(ctx context.Context, loc *UseLocation, uses *v1beta1.TaskSpec) error {
+	areParamsFromTasksResults := make(map[string]bool)
+	pipelineTasksAndFinally := loc.PipelineSpec.Tasks
+	pipelineTasksAndFinally = append(pipelineTasksAndFinally, loc.PipelineSpec.Finally...)
+	for _, pipelineTask := range pipelineTasksAndFinally {
+		params := pipelineTask.Params
+		for _, param := range params {
+			paramValStr := param.Value.StringVal
+			if strings.HasPrefix(paramValStr, "$(tasks.") && strings.Contains(paramValStr, ".results.") && strings.HasSuffix(paramValStr, ")") {
+				areParamsFromTasksResults[param.Name] = true
+			}
+		}
+	}
```

Comment:
Can we reduce the nesting here?

(let's shove it into a function?)

Reply:
done

Reply:
I have also added a test :ok_hand: